docs/disruption_diagram_alternate_structure_examples.md

```diff
@@ -0,0 +1,159 @@
+# Disruption diagram alternate structure examples
+
+The alternate structure factors out some data (e.g. line color) to the top level, and
+leans more on the client to determine the appropriate presentation of symbols, line segments, etc.
+rather than directly telling the client what asset to use for each piece of the diagram.
+
+## Example 1: Shuttle from Boston College to Babcock St, screen at Government Center
+![image](assets/disruption_diagrams/1.png)
+
+```ts
+{
+  effect: "shuttle",
+  line: "green",
+  current_station_slot_index: 11,
+  effect_region_slot_index_range: [0, 8],
+  slots: [
+    {type: "terminal", label_id: "place-lake"},
+    {
+      label: {full: "South Street", abbrev: "South St"},
+      show_symbol: true
+    },
+    {
+      label: {full: "Chestnut Hill Avenue", abbrev: "Chestnut Hill Ave"},
+      show_symbol: true
+    },
+    {
+      label: {full: "Chiswick Road", abbrev: "Chiswick Rd"},
+      show_symbol: true
+    },
+    {
+      label: "…",
+      show_symbol: false
+    },
+    {
+      label: {full: "Griggs Street", abbrev: "Griggs St"},
+      show_symbol: true
+    },
+    {
+      label: {full: "Harvard Avenue", abbrev: "Harvard Ave"},
+      show_symbol: true
+    },
+    {
+      label: {full: "Packard's Corner", abbrev: "Packard's Cn"},
+      show_symbol: true
+    },
+    {
+      label: {full: "Babcock Street", abbrev: "Babcock St"},
+      show_symbol: true
+    },
+    {
+      label: {full: "… via Copley & Kenmore", abbrev: "… via Copley & Kenmore"},
+      show_symbol: false
+    },
+    {
+      label: {full: "Park Street", abbrev: "Park St"},
+      show_symbol: true
+    },
+    {type: "terminal", label_id: "place-gover"}
+  ]
+}
+```
+
+- - -
+
+## Example 2: Suspension from Charles/MGH to South Station, screen at Charles/MGH
+![image](assets/disruption_diagrams/2.png)
+
+(Correction: The first "Downtown Crossing" label should be "Kendall/MIT")
+
+```ts
+{
+  effect: "suspension",
+  line: "red",
+  current_station_slot_index: 2,
+  effect_region_slot_index_range: [3, 4],
```

Comment:
Would this actually be `[3, 5]`? Or is the range in this case just for the icons and the gray lines would be something the client would figure out on its own?

Reply:
Yes, the range is for the slots that should have "disruption" icons—either the shuttle black circles or the suspension/station closure stop signs.

Using indices of the slots rather than the edges ensures consistent behavior even when the range starts at the left end of the line or ends at the right end.